crates/file_finder/src/file_finder.rs

```diff
@@ -204,15 +204,22 @@ impl PartialOrd for ProjectPanelOrdMatch {
 #[derive(Debug, Default)]
 struct Matches {
     separate_history: bool,
+    opened_path: Option<FoundPath>,
     matches: Vec<Match>,
 }
 
 #[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone)]
-enum Match {
+enum MatchData {
     History(FoundPath, Option<ProjectPanelOrdMatch>),
     Search(ProjectPanelOrdMatch),
 }
 
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Clone)]
+struct Match {
+    data: MatchData,
+    history_score: usize,
```

Comment:
I see that all `MatchData::Search` get `no_history_score` here, similar to what the old code did.
Does it really make sense to have an extra struct here, or should we rather move the `history_score` into `MatchData::Search`?

If something needs to access this and it's not very convenient to always match this enum to get the number, we can extract it into a function, that will return a `no_history_score` (`0`) for the `Search` variant.

If we are able to do the latter, we can remove `struct Match` entirely and keep the old name for `enum MatchData` -> `enum Match`.

Reply:
Reverted `struct MatchData` -> `enum Match`, and merged history score into the `History` variant.

`HistoryEntry` struct is there just to simplify `PartialOrd`, `Ord` and `PartialEq` implementations.

The fact that along with the `Ord` implementation there's also a `Matches::cmp_matches` is somewhat confusing, but seems ok to me, since it's different kinds of order defined for the same entity.

Regarding the tests, I still haven't figured out how to check that the selected entry is indeed visible after modifications of the matches list. I've looked into `Picker::logical_scroll_top_index`, but in my tests it returned different indexes from what I've expected, though `UniformList::prepaint` calculated correct visible entries range. Even having the top visible index in place it's not clear to me, how to check that the required index is visible. Any guidance would be highly appreciated.

Looking forward to your feedback.

Reply:
I was indeed, confused by the `Ord` as not sure why it is needed besides deduplication — the comment above does not explain this and by looking at the diff, I've mentioned the only usage but I might be wrong.
If it's indeed about the deduplication only, let's not use any `BTreeSet` at all and do direct binary search and remove `Ord` and anything extra.

I am not certain about testing such things myself, as there's indeed a very `render`-specific code we have to test which might not be even rendering properly as we're testing headless.

So let's keep testing the selections only, I think it's good enough as long as you manually test it to ensure that it works.